src/args.rs

```diff
@@ -77,13 +77,32 @@ fn parse_interface(src: &str) -> Result<IpAddr, std::net::AddrParseError> {
 
 /// Checks wether the auth string is valid, i.e. it follows the syntax username:password
 fn parse_auth(src: &str) -> Result<(String, String), String> {
-    match src.find(':') {
-        Some(_) => {
-            let split = src.split(':').collect::<Vec<_>>();
-            Ok((split[0].to_owned(), split[1].to_owned()))
+    let mut split = src.splitn(2, ':');
+
+    let username = match split.next() {
+        Some(username) => username,
+        None => {
+            return Err(
+                "Invalid credentials string, expected format is username:password".to_owned(),
+            )
         }
-        None => Err("Correct format is username:password".to_owned()),
+    };
+
+    let password = match split.next() {
```

Comment:
Since we can't unroll an iterator directly into a tuple I suppose we're sadly stuck doing it this duplicated way.